### PR TITLE
Remove WarpX::GetInstance call from ParserFilter functor

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR removes the call to `WarpX::GetInstance` from ParserFilter functor. As a consequence, `time` has to be passed as an argument.

### CODE REVIEW

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp [hunk 1]
```diff
@@ -339,7 +339,7 @@ FlushFormatPlotfile::WriteWarpXHeader(
 void
 FlushFormatPlotfile::WriteParticles(const std::string& dir,
                                     const amrex::Vector<ParticleDiag>& particle_diags,
-                                    bool isBTD) const
+                                    const amrex::Real time, bool isBTD) const
```
Comment (NeilZaim) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp:342]: Shouldn't line 102 of this file be modified as well? Maybe there are other instances of this function being called?
Edit: Had a quick look and did not see any other instance

Comment (NeilZaim) [reply Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp:342]: Good news is that the automated tests appear to have caught this :)

Comment (lucafedeli88) [reply Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp:342]: you are totally right! I should have corrected the bug!

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.H [hunk 2]
```diff
@@ -48,10 +48,12 @@ public:
     /** \brief Write particles data to file.
      * \param[in] filename name of output directory
      * \param[in] particle_diags Each element of this vector handles output of 1 species.
+     * \param[in] time the simulation time
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.H]:
```suggestion
     * \param[in] time the simulation time on the coarsest level
```

File: Source/Particles/Filter/FilterFunctors.H [hunk 3]
```diff
@@ -91,14 +90,15 @@ struct ParserFilter
      * \param a_is_active whether the test is active
      * \param a_filter_parser parser taking t, x, y, z, ux, uy, and uz, and returning a boolean for selected particle
      * \param a_mass mass of the particle species
+     * \param time simulation time
```
Comment (ax3l) [inline Source/Particles/Filter/FilterFunctors.H]:
```suggestion
     * \param time simulation time on the coarsest level
```

File: Source/Particles/Filter/FilterFunctors.H [hunk 4]
```diff
@@ -136,10 +136,10 @@ private:
 public:
     /** Parser function with 7 input variables, t,x,y,z,ux,uy,uz */
     amrex::ParserExecutor<7> const m_function_partparser;
-    /** Store physical time. */
-    amrex::ParticleReal m_t;
     /** Mass of particle species */
     amrex::ParticleReal m_mass;
+    /** Store physical time. */
```
Comment (ax3l) [inline Source/Particles/Filter/FilterFunctors.H]:
```suggestion
    /** Store physical time on the coarsest level. */
```

**Review by ax3l — APPROVED** [review]
Thank you! :)

**Pull request merged**